### PR TITLE
feat: onboarding follow sources

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

Add option to follow sources during onboarding.

Initial source recommendations are based on tags you selected:
Screenshot 2024-11-21 at 14 49 40

You can also search:
Screenshot 2024-11-21 at 14 49 46

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-678 #done 

### Preview domain
https://as-678-follow-sources-onboarding.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/onboarding/OnboardingHeader.tsx [hunk 1]
```diff
@@ -53,8 +59,7 @@ export const OnboardingHeader = ({
             position={LogoPosition.Relative}
             linkDisabled
           />
-          {(activeScreen === OnboardingStep.EditTag ||
-            activeScreen === OnboardingStep.ContentTypes) && (
+          {showButtonOnScreens.includes(activeScreen) && (
```
Comment (rebelchris) [inline packages/shared/src/components/onboarding/OnboardingHeader.tsx:62]: More readable in my opinion

File: packages/shared/src/components/onboarding/Sources/Sources.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+          valueChanged={onSearch}
+        />
+        <div
+          role="list"
+          aria-busy={isPending}
+          className="flex flex-row flex-wrap justify-center gap-4"
+        >
+          {isPending &&
+            placeholderSources.map((item) => (
+              <ElementPlaceholder key={item} className="h-10 rounded-12">
```
Comment (rebelchris) [inline packages/shared/src/components/onboarding/Sources/Sources.tsx:175]: Wrong one

File: packages/shared/tailwind.config.ts [hunk 3]
```diff
@@ -148,6 +148,7 @@ export default {
       6: '0.375rem',
       8: '0.5rem',
       10: '0.625rem',
+      11: '0.6875rem',
```
Comment (rebelchris) [inline packages/shared/tailwind.config.ts:151]: Needed for inner border, else @omBratteng would get angry

Comment (omBratteng) [reply packages/shared/tailwind.config.ts:151]: 😂

**Comment by omBratteng** [comment]
https://github.com/user-attachments/assets/c14d7772-ef82-4301-b41e-0a87fd3f88a3

It sometimes falls back to initial data while it's pending.
And search input field changes width
image
image

**Review by omBratteng — CHANGES_REQUESTED** [review]
See the comment with video

**Comment by omBratteng** [comment]
Tested in preview, and it works fine

File: packages/shared/src/components/onboarding/Sources/Sources.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  const [searchQuery, setSearchQuery] = React.useState<string>();
+  const [onSearch] = useDebounceFn(setSearchQuery, 200);
+
+  const { feedSettings } = useFeedSettings();
+
+  const { onFollowSource, onUnfollowSource } = useTagAndSource({
+    origin: Origin.Onboarding,
+  });
+
+  const selectedTags = feedSettings?.includeTags || ['webdev'];
```
Comment (rebelchris) [inline packages/shared/src/components/onboarding/Sources/Sources.tsx:39]:
```suggestion
  const selectedTags = feedSettings?.includeTags || [];
```

**Review by omBratteng — APPROVED** [review]

**Pull request merged**